[ADD] website: menus page (#5036)
odoo/documentation: content/applications/websites/website/pages/menus.rst

=====
Menus
=====

> Menus are used to organize your website’s content effectively and help visitors navigate through
> your web pages effectively. User-friendly and well-structured website menus also play a crucial role

Reviewer: ```suggestion
Menus are used to organize your website’s content and help visitors navigate through
your web pages effectively. User-friendly and well-structured website menus also play a crucial role
```
Repetition of "effectively"

in improving :doc:`search engine rankings <seo>`.

Odoo allows you to customize the content and appearance of these menus to your needs.

Menu editor
===========

To edit your website's menu, go to :menuselection:`Website --> Site --> Menu Editor`. From there,

> you can:

Reviewer 2: Imo this is not necessary since the section covers the basic features of the menu editor, but we can talk about it :)

- rename a menu item or change its URL using the :guilabel:`Edit Menu Item` icon;
- delete a menu item using the :guilabel:`Delete Menu Item` icon;
- move a menu item by drag and dropping it to the desired place in the menu;
- create a drop-down menu by drag and dropping the sub-menu items to the right, underneath their

>   parent menu.

Reviewer: Capitals to start the sentences?
I think I would put the actions (rename, delete, move a menu item, create a drop down menu) in bold.

Reviewer 2: I put the actions in bold, but didn't use capital letters as each list item continues the sentence "From there, you can:"

.. image:: menus/menu-editor.png
   :alt: Menu editor with sub-menus

.. note::
  - By default, pages are added to the menu when :doc:`they are created <../pages>`.
  - You can also access the menu editor by clicking :guilabel:`Edit`, then selecting any menu item

>     and clicking the :guilabel:`Edit menu` icon.

Reviewer: I don't get it...

Reviewer 2: I added a screenshot here to clarify

Creating menu items
-------------------

To create a menu item, click :guilabel:`Add Menu Item`, then, in the pop-up window that appears on

> the screen, enter the :guilabel:`Name` and :guilabel:`Url` of the related page. Click :guilabel:`OK`

Reviewer: ```suggestion
the screen, enter the :guilabel:`Name` and :guilabel:`Url or Email` of the related page. Click :guilabel:`OK`
```

Reviewer 2: I know what you mean here since this is the actual label of the field but I think it doesn't make much sense to mention "email" in the context of the sentence. Would removing the guilabel solve this issue here?

to create the menu item.

.. tip::
   Type `/` in the :guilabel:`URL` field to search for an existing page of your website.

Mega menus
----------

Mega menus are similar to drop-down menus, but instead of a simple list of sub-menus, they display a
panel divided into groups of navigation options. This makes them suitable for websites with large

> amounts of contents, as they can help include all of your web pages in the menu, while still making

Reviewer: ```suggestion
amounts of content, as they can help include all your web pages in the menu while making
```

all menu options visible at once. Mega menus can also be structured more visually than regular

> drop-down menu items, for example through layout, typography and icons.

Reviewer: ```suggestion
drop-down menu items, for example, through layout, typography, and icons.
```

.. image:: menus/mega-menu.png
   :alt: Mega menu in the navigation bar.

To create a mega menu, proceed as follows:

#. Go to :menuselection:`Website --> Site --> Menu Editor`, click :guilabel:`Add Mega Menu Item`,
   then enter a :guilabel:`Name`, and click :guilabel:`OK`.

> #. Click the mega menu in the navigation bar, then click :guilabel:`Edit`.

Reviewer: What navigation bar? Should I leave the menu editor? Not clear to me.

   - Edit its text directly in the building block;
   - Edit the link by clicking the :guilabel:`Edit link` button in the small preview popup, then
     enter type `/` to search for a page on your website, or `#` to search for an existing custom
     anchor.

  .. image:: menus/mega-menu-option.png
     :alt: Edit a mega menu option

.. note::
   Each mega menu option is an individual building block, which means you can:

   - move it by drag and dropping the block to the desired position in the mega menu;
   - delete it by deleting the block;
   - edit its text directly in the related block and format it using the :guilabel:`Customize` tab
     in the website builder.

Customization
=============

To customize the appearance of your website's menu, click :guilabel:`Edit` and select a menu item.
You can then adapt the fields in the :guilabel:`Customize` tab, for example, change the
:guilabel:`Template` for the header and adapt the appearance of the navigation bar.

To adapt the general layout of a mega menu, select the mega menu, then go the the
:guilabel:`Mega menu` section in the website builder and select the desired :guilabel:`Template` and

> :guilabel:`Size`. Since mega menus are composed of building blocks, you can also customize each

Reviewer: ```suggestion
:guilabel:`Size`. Since mega menus are composed of building blocks, you can customize each
```

- **delete** a menu item using the :guilabel:`Delete Menu Item` icon;
- **move** a menu item by dragging and dropping it to the desired place in the menu;
- **create a regular drop-down menu** by dragging and dropping the sub-menu items to the right,
  underneath their parent menu.

.. image:: menus/menu-editor.png
   :alt: Menu editor with sub-menus

.. note::

  You can also access the menu editor by clicking :guilabel:`Edit`, selecting any menu item and
  clicking the :guilabel:`Edit menu` icon.

  .. image:: menus/edit-menu-icon.png
     :alt: Access the Menu editor while in Edit mode.

Adding regular menu items
-------------------------

By default, pages are added to the menu as regular menu items when :doc:`they are created
<../pages>`. You can also add regular menu items from the menu editor by clicking :guilabel:`Add
Menu Item`. Enter the :guilabel:`Name` and URL of the related page in the pop-up window that appears
on the screen and click :guilabel:`OK`.

.. tip::
   In the :guilabel:`URL or Email` field, you can type `/` to search for a page on your website or
   `#` to search for an existing custom anchor.

> Adding mega menus

Reviewer 3: Maybe just `Mega menus` since it's about adding and editing their appearance in the website builder?

Or change the structure, for example:

```restructuredtext
=====
Menus
=====

Menu editor
===========

Regular menu
============

Mega menu
=========

Header and navigation bar
=========================
```

Reviewer 2: I think I want to keep it as such as 1) it's still about *adding* menu items (for both regular and mega menus), 2) you can't really add a mega menu without customizing it (e.g. changing the URL and name of the options) - well, you can but IMO that would be useless :) - and 3) to me, both belong in the menu editor part as it is the "starting point" for both. Does that make sense to you?

-----------------

Mega menus are similar to drop-down menus, but instead of a simple list of sub-menus, they display a
panel divided into groups of navigation options. This makes them suitable for websites with large
amounts of content, as they can help include all of your web pages in the menu while still making
all menu options visible at once. Mega menus can also be structured more visually than regular
drop-down menu items, for example, through layout, typography, and icons.

.. image:: menus/mega-menu.png
   :alt: Mega menu in the navigation bar.

To create a mega menu, go to :menuselection:`Website --> Site --> Menu Editor` and click
:guilabel:`Add Mega Menu Item`. Enter the :guilabel:`Name` of the mega menu in the pop-up, click
:guilabel:`OK`, then :guilabel:`Save`.

To adapt the options and layout of the mega menu, click it in the navigation bar, then click
:guilabel:`Edit`. Mega menus are composed of building blocks, which means you can customize each
component individually using inline formatting, as well as the options available in the
:guilabel:`Customize` tab in the website builder. For example, you can:

- edit the text directly in the building block;

> - edit an option's URL by selecting the option, then clicking the :guilabel:`Edit link` button in

Reviewer 3: ```suggestion
- edit an option's URL by selecting the option, then clicking the :guilabel:`Edit Link` button in
```
I assume an "option" is a sub-item in a mega menu, but the word is vague. Consider changing it, even if I guess you found the word used somewhere in Odoo :)

Reviewer 2: Changed it, and I also changed it in the next bullet points (move and delete)

  the small preview pop-up. Type `/` to search for a page on your website, or `#` to search for an
  existing custom anchor.

  .. image:: menus/mega-menu-option.png
     :alt: Edit a mega menu option.

- move an option by dragging and dropping the related block to the desired position in the mega
  menu;
- delete an option by deleting the related block;
- adapt the general layout of the mega menu by selecting the desired :guilabel:`Template` and
  :guilabel:`Size` in the :guilabel:`Mega menu` section in the :guilabel:`Customize` tab.

> Header and navigation bar appearance

Reviewer 3: _NB: the comment is made assuming you leave the structure as it is:_

I would make this a `.. tip::` since it's not solely related to the menu + there isn't enough content to justify an H2 heading, and I don't think it should be on the same level as the menu editor.
If you still would like to keep the heading, you could add a screenshot to make the section longer, maybe showing some of the customization options available.
